### ISSUE TITLE
Az webapp deployment slot log or status - feature request

### ISSUE THREAD

**Issue by 4ttedertlf** [Azure/azure-cli#22953]
**Related command**

az webapp deployment slot swap - is the related command

**Is your feature request related to a problem? Please describe.**

I can't swap the slots because the source slot is not resolving a ping. I can see that in the phone app for Azure but I can't see that in the Azure portal log or with Az CLI (at least that I could find).

I need a way to determine the issue with a failed swap, ie its previous or current state, in order to know if I can swap - when using automation or CICD tooling. 

**Describe the solution you'd like**

I would like to know the current status of the slot, and the last failed action that might prevent a swap. 

**Describe alternatives you've considered**

All I can think of is to redeploy but the issue is I've redeployed 3 times and still can't swap. If deployment succeeded but I can't swap, there is some middle piece of information I need but don't have. 

**Additional context**

- App Service for Linux and Node 14, 2 slots (production and stage)
- az cli command used in GitHub action with Az CLI action command: az webapp deployment slot swap --name 4tt-1 --resource-group basic-express-typescript --slot stage

**Comment by yonzhan** [comment]
route to CXP team

**Comment by 4ttedertlf** [comment]
@diberry - fyi

**Comment by diberry** [comment]
@cephalin FYI - we discussed this as part of the deployment meeting.

**Comment by panchagnula** [comment]
> @cephalin FYI - we discussed this as part of the deployment meeting.

@diberry what was the discussion / conclusion here?

**Comment by diberry** [comment]
@panchagnula 

The discussion was that a slot can get into an unknown state or the swap can fail, there isn't any help from the Azure CLI to debug that situation. The conclusion was that the Azure CLI should be able to report in some way to be able to debug a slot state.

**Comment by panchagnula** [comment]
> @panchagnula
> 
> The discussion was that a slot can get into an unknown state or the swap can fail, there isn't any help from the Azure CLI to debug that situation. The conclusion was that the Azure CLI should be able to report in some way to be able to debug a slot state.

Thanks- CLI can report if there is an API for us to use. From reading this issue, this is not CLI specific & occurs on portal as well. Hence, a backend. approach, may be exposing in logs will help all clients surface the issue to the user.

**Comment by alexwenzel** [comment]
+1

**Comment by kumy** [comment]
Is there any workaround to get slot logs and status? :pray:

**Comment by seligj95** [comment]
> [!NOTE]
> **[AI-Assisted Triage]** This comment was generated with AI assistance as part of an App Service CLI issue review. Reply to this comment and tag @seligj95 to follow up.

**Summary**: Request for swap status and log visibility through CLI — currently no way to monitor swap progress or view swap logs.

**Status**: On Hold

**Priority**: P3

**Next Steps**: On hold — blocked on platform API support for swap status and logs. Will resume once the platform exposes the necessary APIs.
